File: app/src/main/java/com/berkatfaatulohalawa1711010164/facevoting/helper/ServiceError.java
package com.berkatfaatulohalawa1711010164.facevoting.helper;

import com.berkatfaatulohalawa1711010164.facevoting.config.Constants;

import okhttp3.OkHttpClient;
import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

public class ServiceError {
    public static Retrofit.Builder builder =
            new Retrofit.Builder()
                    .baseUrl(Constants.URL)
                    .addConverterFactory(GsonConverterFactory.create());

    public static Retrofit retrofit = builder.build();

    public static OkHttpClient.Builder httpClient =
            new OkHttpClient.Builder();

    public static <S> S createService(
            Class<S> serviceClass) {
        return retrofit.create(serviceClass);
    }
}

File: android/facevoting/app/src/main/java/com/berkatfaatulohalawa1711010164/facevoting/helper/ServiceError.kt
package com.berkatfaatulohalawa1711010164.facevoting.helper

import com.berkatfaatulohalawa1711010164.facevoting.config.Constants
import okhttp3.OkHttpClient
import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory

object ServiceError {
    val builder: Retrofit.Builder = Retrofit.Builder()
        .baseUrl(Constants.URL)
        .addConverterFactory(GsonConverterFactory.create())

    val retrofit: Retrofit = builder.build()

    val httpClient: OkHttpClient.Builder = OkHttpClient.Builder()

    fun <S> createService(serviceClass: Class<S>): S = retrofit.create(serviceClass)
}
